flask_jwt_extended/view_decorators.py

```diff
@@ -170,25 +171,36 @@ def _decode_jwt_from_headers():
     header_type = config.header_type
 
     # Verify we have the auth header
-    jwt_header = request.headers.get(header_name, None)
-    if not jwt_header:
+    auth_header = request.headers.get(header_name, None)
+    if not auth_header:
         raise NoAuthorizationError("Missing {} Header".format(header_name))
 
     # Make sure the header is in a valid format that we are expecting, ie
     # <HeaderName>: <HeaderType(optional)> <JWT>
+    jwt_header = None
+
+    # Check if header is comma delimited, ie
+    # <HeaderName>: <field> <value>, <field> <value>, etc...
+    if header_type:
+        field_values = split(r',\s*', auth_header)
+        jwt_header = [s for s in field_values if s.startswith(header_type)]
```

Comment:
I think this could break if we had two similiar headers in here, ie `Bearer` and `Bearer2`. Although that isn't likely to happen, I wonder if we could do this check via something like `s.split[0] == header_type` instead?

Reply:
Yeah I wasn't thinking about that, or maybe just `s.startswith(header_type+' ')`